pkg/controller/sandbox/sandbox_controller.go

```diff
@@ -176,18 +197,25 @@ func (r *SandboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		utils.SetSandboxCondition(newStatus, cond)
 	}
 
-	args := core.EnsureFuncArgs{Pod: pod, Box: box, NewStatus: newStatus}
 	switch newStatus.Phase {
 	case agentsv1alpha1.SandboxPending:
+		if box.Annotations[agentsv1alpha1.SandboxHashWithoutImageAndResources] == "" {
+			clone := box.DeepCopy()
+			body := fmt.Sprintf(`{"metadata":{"annotations":{"%s":"%s"}}}`, agentsv1alpha1.SandboxHashWithoutImageAndResources, hashWithoutImageAndResource)
+			if err = r.Patch(context.TODO(), clone, client.RawPatch(types.MergePatchType, []byte(body))); err != nil {
```

Comment:
plz assign the updated sandbox object (clone) to the box variable, otherwise, the EnsureSandboxPhasePending will have update conflict in the first reconcile